src/main/java/com/ncc/aif/AIFUtils.java

```diff
@@ -1117,11 +1121,67 @@ public static Resource linkToExternalKB(Model model, Resource toLink, String ext
                 .build();
     }
 
+    /**
+     * This inner class encapsulates the LDC representation of time.
+     */
+    public static final class LDCTimeComponent {
+        public enum LDCTimeType { ON, BEFORE, AFTER, UNKNOWN }
+
+        private final LDCTimeType type;
+        private final String year;
+        private final String month;
+        private final String date;
+
+        public LDCTimeComponent(LDCTimeType type, String year, String month, String date) {
+            this.type = type;
+            this.year = year;
+            this.month = month;
+            this.date = date;
```

Comment:
I would update the name of the `date` parameter to `day` so it is not confusing that this field actually represents a day.